core/src/Revolution/Processors/Security/User/Create.php

```diff
@@ -232,6 +233,52 @@ public function sendNotificationEmail() {
                 'html' => true,
             ]);
         }
+
+        if ($this->getProperty('passwordgenmethod') === 'user_email_specify' && $this->modx->getService('hashing', modHashing::class)) {
+            $activationHash = $this->modx->hashing->getHash('md5', 'hashing.modMD5', [])->hash($this->object->get('email') . '/' . $this->object->get('id'));
```

Comment:
This is **not** a secure hash/tokens and introduces a potential vulnerability. The scope would be limited because the "tokens" have a limited lifetime after manually being created by an admin, however it makes the hash _very_ predictable.

In 3.x we can use `random_bytes`, so I would suggest:

```` 
$activationHash = bin2hex(random_bytes(32));
```` 

for a 64-character secure random token.